src/main/kotlin/net/rrerr/minetrix/MessageFormatter.kt

```diff
@@ -0,0 +1,111 @@
+package net.rrerr.minetrix
+
+import de.msrd0.matrix.client.event.FormattedTextMessageContent
+import de.msrd0.matrix.client.event.MessageContent
+import net.md_5.bungee.api.ChatColor
+import net.md_5.bungee.api.chat.BaseComponent
+import net.md_5.bungee.api.chat.ClickEvent
+import net.md_5.bungee.api.chat.TextComponent
+import org.w3c.dom.Document
+import org.w3c.dom.NodeList
+import org.xml.sax.InputSource
+import java.io.StringReader
+import javax.xml.parsers.DocumentBuilderFactory
+
+
+class MessageFormatter(var message: MessageContent) {
+    companion object {
+        val TAGS = mapOf(
+            "B" to ChatColor.BOLD,
+            "STRONG" to ChatColor.BOLD,
+            "EM" to ChatColor.ITALIC,
+            "I" to ChatColor.ITALIC,
+            "U" to ChatColor.UNDERLINE,
+            "DEL" to ChatColor.STRIKETHROUGH
+        )
+    }
+
+    val urlPattern = Regex("(https?://)?\\S+\\.\\S+(/|/\\S)?")
+    val urlStart = Regex("^https?://.*")
```

Comment:
The `.*` in the regex here is useless for what it's used (a check for the start of the text).  Can be simplified by `^https?://`